src/LogReader.ts

```diff
@@ -62,15 +66,13 @@ export class LogReader implements Disposable {
         const docs: readonly TextDocument[] = workspace.textDocuments;
 
         let needsSync: boolean = false;
-        for (const doc of docs) {
-            if (!needsSync && Utilities.isSarifFile(doc)) {
-                needsSync = true;
-            }
+        for (const doc of docs.filter((doc) => Utilities.isSarifFile(doc))) {
```

Comment:
Why walk through all once you figured out that you need sync?

Reply:
Not sure I follow. The purpose here is also to `read()`.

Reply:
This actually demonstrates perfectly why optional parameters and booleans are not a good choice.

the "read" function takes an optional boolean that indicates whether the information should be "synced" to VSCode's diagnostic collection, which obviously we don't want during import.

Since the function has this parameter as an "optional" bool, from this code, you cannot tell that it is indeed "not" synching to VSCode's diagnostics and hence a call to "sync diagnostics" needs to be made when complete.

I'm fixing this so that it is clear.

---
In reply to: [405168654](https://github.com/microsoft/sarif-vscode-extension/pull/218#discussion_r405168654) [](ancestors = 405168654)